python/ray/air/integrations/wandb.py

```diff
@@ -0,0 +1,582 @@
+# Hook that is invoked after running wandb.init in WandbLoggerCallback
+# to process information about the W&B run.
+# It takes in a W&B run object and doesn't return anything.
+# Example: "your.module.wandb_process_run_info_hook".
+WANDB_PROCESS_RUN_INFO_HOOK = "WANDB_PROCESS_RUN_INFO_HOOK"
+_VALID_TYPES = (
+    Number,
+    wandb.data_types.Audio,
+    wandb.data_types.BoundingBoxes2D,
+    wandb.data_types.Graph,
+    wandb.data_types.Histogram,
+    wandb.data_types.Html,
+    wandb.data_types.Image,
+    wandb.data_types.ImageMask,
+    wandb.data_types.Molecule,
+    wandb.data_types.Object3D,
+    wandb.data_types.Plotly,
+    wandb.data_types.Table,
+    wandb.data_types.Video,
+)
+_VALID_ITERABLE_TYPES = (
+    wandb.data_types.Audio,
+    wandb.data_types.BoundingBoxes2D,
+    wandb.data_types.Graph,
+    wandb.data_types.Histogram,
+    wandb.data_types.Html,
+    wandb.data_types.Image,
+    wandb.data_types.ImageMask,
+    wandb.data_types.Molecule,
+    wandb.data_types.Object3D,
+    wandb.data_types.Plotly,
+    wandb.data_types.Table,
+    wandb.data_types.Video,
+)
+
+
+@PublicAPI(stability="alpha")
+def setup_wandb(
+    config: Optional[Dict] = None, rank_zero_only: bool = True, **kwargs
+) -> Union[Run, RunDisabled]:
+    """Set up a Weights & Biases session.
+
+    This function can be used to initialize a Weights & Biases session in a
+    (distributed) training or tuning run.
+
+    By default, the run ID is the trial ID, the run name is the trial name, and
+    the run group is the experiment name. These settings can be overwritten by
+    passing the respective arguments as ``kwargs``, which will be passed to
+    ``wandb.init()``.
+
+    In distributed training with Ray Train, only the zero-rank worker will initialize
+    wandb. All other workers will return a mock object, so that logging is not
+    duplicated in a distributed run. This can be disabled by passing
+    ``rank_zero_only=False``, which will then initialize wandb in every training
+    worker.
+
+    The ``config`` argument will be passed to Weights and Biases and will be logged
+    as the run configuration. If wandb-specific settings are found, they will
+    be used to initialize the session. These settings can be
+
+    - api_key_file: Path to locally available file containing a W&B API key
+    - api_key: API key to authenticate with W&B
+
+    If no API information is found in the config, wandb will try to authenticate
+    using locally stored credentials, created for instance by running ``wandb login``.
+
+    All other keys found in the ``wandb`` config parameter will be passed to
+    ``wandb.init()``. If the same keys are present in multiple locations, the
+    ``kwargs`` passed to ``setup_wandb()`` will take precedence over those passed
+    as config keys.
+
+    Args:
+        config: Configuration dict to be logged to weights and biases. Can contain
+            arguments for ``wandb.init()`` as well as authentication information.
```

Comment:
Can use two different arguments for the two different purposes, rather than overloading onto the same arg?

Reply:
This is the API as it exists today so if we change this we may have to do this gracefully.

I'd be open to adding a `init_kwargs` subkey that will contain the kwargs so that they are not on the same level.

Should we defer this to a follow-up PR? At the moment we're just using the existing config API.

Reply:
Ok that should be fine...though to be fair we are changing the API so backwards compatibility is not a requirement

Reply:
But I would say it's better to do this sooner than later to avoid churning on the API too much

Reply:
Let's keep it as is for now, as it's the same API for the wandb logger callback and we would have to change both.